### PR TITLE
Extent refactoring and dumpability metadata

### PR DESCRIPTION
**Pull request by davidtgoldblatt**
These commits add support for setting regions of the address space undumpable (on supported platforms).

Currently, this isn't "plugged-in" anywhere; nothing makes pages undumpable. However, this support is required in order to implement some of the subsequent large-address-space ideas we have.

Along the way, I added some extent refactoring / documentation.

### CODE REVIEW

File: include/jemalloc/internal/extent_structs.h [hunk 1]
```diff
@@ -23,13 +23,14 @@ struct extent_s {
...
 	 * c: committed
+	 * d: dumpable
 	 * z: zeroed
 	 * t: state
 	 * i: szind
 	 * f: nfree
 	 * n: sn
 	 *
-	 * nnnnnnnn ... nnnnnfff fffffffi iiiiiiit tzcbaaaa aaaaaaaa
+	 * nnnnnnnn ... nnnnnffff ffffffii iiiiiitt zdcbaaaa aaaaaaaa
```
Comment (djwatson) [inline include/jemalloc/internal/extent_structs.h]: groupings are no longer all 8, on purpose?

Comment (davidtgoldblatt) [reply include/jemalloc/internal/extent_structs.h]: Woops, nope.

Comment (davidtgoldblatt) [reply include/jemalloc/internal/extent_structs.h]: Fixed.

**Comment by davidtgoldblatt** [comment]
Only comments have changed since last green build - merging.

**Pull request merged**